lib/providers/quoters/RfqQuoter.ts

```diff
@@ -9,21 +9,36 @@ import { PostQuoteResponseJoi } from '../../handlers/quote';
 import { log } from '../../util/log';
 import { metrics } from '../../util/metrics';
 import { generateRandomNonce } from '../../util/nonce';
+import { PortionProvider } from '../portion/PortionProvider';
 import { Quoter, QuoterType } from './index';
 
 export class RfqQuoter implements Quoter {
   static readonly type: QuoterType.UNISWAPX_RFQ;
 
-  constructor(private rfqUrl: string, private serviceUrl: string, private paramApiKey: string) {}
+  constructor(
+    private rfqUrl: string,
+    private serviceUrl: string,
+    private paramApiKey: string,
+    private portionProvider: PortionProvider
+  ) {}
 
   async quote(request: DutchRequest): Promise<Quote | null> {
     if (request.routingType !== RoutingType.DUTCH_LIMIT) {
       log.error(`Invalid routing config type: ${request.routingType}`);
       return null;
     }
 
+    const portion = (await this.portionProvider.getPortion(request.info, request.info.tokenIn, request.info.tokenOut))
+      .portion;
+
     const swapper = request.config.swapper;
     const now = Date.now();
+    // we must adjust up the exact out swap amount to account for portion, before requesting quote from RFQ quoter
+    const portionAmount =
+      portion?.bips && request.info.type === TradeType.EXACT_OUTPUT
+        ? request.info.amount.mul(portion?.bips)
+        : undefined;
+    const amount = portionAmount ? request.info.amount.add(portionAmount) : request.info.amount;
```

Comment:
This is a **big** change team please make sure everyone is comfortable with this:

@marktoda @rileydcampbell @zhongeric @unicorn-here

Reply:
yep this seems right to me, we should we requesting RFQ for the amount + portion